NDTensors/test/test_blocksparse.jl

```diff
@@ -284,39 +284,44 @@ using Test: @test, @test_throws, @testset
       A = dev(BlockSparseTensor{elt}([(2, 1), (1, 2)], [2, 2], [2, 2]))
       randn!(A)
       U, S, V = svd(A)
-      @test @allowscalar array(U) * array(S) * array(V)' ≈ array(A)
+      Asvd = contract(contract(U, (1, -2), S, (-2, 3)), (1, -2), V, (3, -2))
```

Comment:
@mtfishman Here U is a blocksparse tensor and S is a diagblocksparse tensor so this now tests the GPU BS * DiagBS. But I guess it doesn't check GPU BS * UniformDiag

Reply:
I think it is better to keep these tests how they were, since then they are testing the SVD more in isolation and not testing both SVD and contract. Could you write devoted tests for block sparse (uniform) diag * dense contraction?

Reply:
Makes sense, I can do that. I found an issue with directly contracting block sparse with block sparse uniform diag when trying to make a test and reported it here https://github.com/ITensor/ITensors.jl/issues/1474.

Reply:
Hopefully it is easy enough to find an example contraction that works and circumvents #1474, maybe that issue is particular to inner product-like contractions and a matrix multiplication-like contraction would work.

Reply:
See https://github.com/ITensor/ITensors.jl/issues/1474#issuecomment-2138447085 and #1475.

Reply:
@kmp5VT thanks for reporting issues you're finding with DiagBlockSparse contraction, for the sake of this PR I think we can just make a test based on one of the DiagBlockSparse contractions that works and worry about the ones that are failing in future PRs.